File: app/Http/Controllers/Api/V1/DollyActionController.php
<?php

namespace App\Http\Controllers\Api\V1;

use App\Http\Controllers\Controller;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;

/**
 * D11 — actions en masse sur les éléments d'un chariot (`GET /dollies/action`).
 *
 * ⚠️ TODO DOCUMENTÉ — partiellement porté.
 *
 * Le contrôleur Blade `DollyActionController` est un routeur de formulaires qui
 * renvoie des vues, avec plus de 25 combinaisons catégorie × action (dates,
 * priorité, type, niveau, statut, conteneur, activité, support, export SEDA/PDF…).
 *
 * Actions portées depuis le 2026-08-05 (points d'entrée explicites, logique simple) :
 *   - ajout/retrait d'éléments : `DollyController` `add-record` / `remove-record`,
 *     `add-communication` / `remove-communication`, `add-slip` / `remove-slip`
 *     (et les autres catégories déjà exposées) ;
 *   - `clear` (équivalent des `clean` du Blade : détache les pivots sans supprimer
 *     les entités) et `rename` (renommage du chariot).
 *
 * Actions NON portées (501), avec la raison technique :
 *   - **`*Delete`** (suppression physique des entités du chariot, ex. `mailDelete`)
 *     : destructif, hors du périmètre d'édition de l'API phase 1 — à réexposer par une
 *     action dédiée après relecture métier.
 *   - **mass-edit en masse** (dates, priorité, type, niveau, statut, conteneur,
 *     activité, support, salle/étagère) : encodent des colonnes incohérentes avec le
 *     schéma actuel — `MailDateChange` écrit `mails.date_exact` (colonne absente),
 *     `MailTypeChange` écrit `mails.type_id` (colonne absente), `MailArchivedChange`
 *     écrit `mails.is_achived` (typo de `is_archived`), `ContainerShelfchange` itère
 *     `slipRecords` au lieu des conteneurs, la relation « shelves » du modèle est
 *     `shelve()`. Un portage approximatif encoderait ces bugs : relecture métier
 *     préalable, chaque édition en masse doit être repensée en point d'entrée dédié.
 *   - **exports** (SEDA XML, inventaire PDF) : classe E2 (phase 3) — exports de fichiers.
 *
 * Le GET `/dollies/action` du Blade (routeur de formulaires) reste non exposé ; les
 * actions portées ont leurs propres routes explicites.
 */
class DollyActionController extends Controller
{
    /**
     * GET /api/v1/dollies/action
     */
    public function index(Request $request): JsonResponse
    {
        return response()->json([
            'type' => 'about:blank',
            'title' => 'Non implémenté',
            'status' => 501,
            'detail' => 'Le routeur d\'actions en masse de DollyActionController n\'est pas exposé : utiliser les points d\'entrée explicites add-*/remove-*, clear et rename (voir l\'en-tête).',
        ], 501);
    }
}

File: backend/src/main/java/com/shelve/dolly/controller/DollyActionController.java
package com.shelve.dolly.controller;

import com.shelve.exception.ApiException;
import com.shelve.security.AuthenticatedUser;
import com.shelve.security.CurrentUser;
import com.shelve.security.Policy;
import java.util.LinkedHashMap;
import java.util.Map;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(value = {"/api/v1/dollies/action"})
public class DollyActionController {
  @GetMapping
  public Map<String, Object> index() {
    AuthenticatedUser auth = CurrentUser.get();
    Policy.check(auth, "dolly_viewAny");
    LinkedHashMap<String, Object> body = new LinkedHashMap<String, Object>();
    body.put("type", "about:blank");
    body.put("title", "Non impl\u00e9ment\u00e9");
    body.put("status", 501);
    body.put(
        "detail",
        "Le routeur d'actions en masse de DollyActionController n'est pas expos\u00e9 : utiliser"
            + " les points d'entr\u00e9e explicites add-*/remove-*, clear et rename.");
    throw new ApiException(HttpStatus.NOT_IMPLEMENTED, "Non impl\u00e9ment\u00e9");
  }
}
